### PR TITLE
[chore] Upgrade kube version to v1.18.3

### PR DESCRIPTION
**Pull request by rahulchheda**
## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [x] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Issues

- #XXX

## Test Plan

- [ ] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -27,7 +27,7 @@ jobs:
       install:
       - make go-mod-download
       before_script:
-      - docker pull kindest/node:v1.17.2
+      - docker pull kindest/node:v1.18.2
```
Comment (pavannd1) [inline .travis.yml:30]: Should this also be 1.18.3 as well?

Comment (rahulchheda) [reply .travis.yml:30]: https://hub.docker.com/r/kindest/node/tags
No tag for 1.18.3

**Comment by pavannd1** [comment]
Still have around 30+ files to go through. Looks good so far.

File: pkg/param/param_test.go [hunk 2]
```diff
@@ -567,19 +570,27 @@ func (s *ParamsSuite) TestProfile(c *C) {
...
+					},
 				},
 			},
 		},
 	}
+	// asList := &crv1alpha1.ActionSetList{
+	// 	Items: []*crv1alpha1.ActionSet{
+	// 		as,
+	// 	},
+	// }
```
Comment (pavannd1) [inline pkg/param/param_test.go:585]: Can this be removed?

File: pkg/resource/resource_test.go [hunk 3]
```diff
@@ -76,28 +76,28 @@ func (s *ResourceSuite) TestActionSetClient(c *C) {
...
 	c.Assert(as, NotNil)
 
-	as2, err := cli.ActionSets(s.namespace).Get(name, emptyGetOptions)
+	as2, err := cli.ActionSets(s.namespace).Get(ctx, name, emptyGetOptions)
 	c.Assert(err, IsNil)
 	c.Assert(as1, DeepEquals, as2)
 
 	as2.Spec = &crv1alpha1.ActionSetSpec{}
-	as3, err := cli.ActionSets(s.namespace).Update(as2)
+	as3, err := cli.ActionSets(s.namespace).Update(context.TODO(), as2, metav1.UpdateOptions{})
```
Comment (pavannd1) [inline pkg/resource/resource_test.go:88]:
```suggestion
	as3, err := cli.ActionSets(s.namespace).Update(ctx, as2, metav1.UpdateOptions{})
```

File: pkg/resource/resource_test.go [hunk 4]
```diff
@@ -76,28 +76,28 @@ func (s *ResourceSuite) TestActionSetClient(c *C) {
...
 	c.Assert(err, IsNil)
 	c.Assert(as1.Spec, IsNil)
 	c.Assert(as3.Spec, NotNil)
 
-	as4, err := cli.ActionSets(s.namespace).Get(name, emptyGetOptions)
+	as4, err := cli.ActionSets(s.namespace).Get(context.TODO(), name, emptyGetOptions)
```
Comment (pavannd1) [inline pkg/resource/resource_test.go:93]:
```suggestion
	as4, err := cli.ActionSets(s.namespace).Get(ctx, name, emptyGetOptions)
```

File: pkg/resource/resource_test.go [hunk 5]
```diff
@@ -76,28 +76,28 @@ func (s *ResourceSuite) TestActionSetClient(c *C) {
...
 	c.Assert(err, IsNil)
 	c.Assert(as4, DeepEquals, as3)
 
-	err = cli.ActionSets(s.namespace).Delete(name, nil)
+	err = cli.ActionSets(s.namespace).Delete(context.TODO(), name, metav1.DeleteOptions{})
```
Comment (pavannd1) [inline pkg/resource/resource_test.go:97]:
```suggestion
	err = cli.ActionSets(s.namespace).Delete(ctx, name, metav1.DeleteOptions{})
```

File: pkg/resource/resource_test.go [hunk 6]
```diff
@@ -76,28 +76,28 @@ func (s *ResourceSuite) TestActionSetClient(c *C) {
...
 	c.Assert(err, IsNil)
 
-	_, err = cli.ActionSets(s.namespace).Get(name, emptyGetOptions)
+	_, err = cli.ActionSets(s.namespace).Get(context.TODO(), name, emptyGetOptions)
```
Comment (pavannd1) [inline pkg/resource/resource_test.go:100]:
```suggestion
	_, err = cli.ActionSets(s.namespace).Get(ctx, name, emptyGetOptions)
```

File: pkg/resource/resource_test.go [hunk 7]
```diff
@@ -117,27 +117,27 @@ func (s *ResourceSuite) TestBlueprintClient(c *C) {
...
 	c.Assert(bp, NotNil)
 
-	bp2, err := cli.Blueprints(s.namespace).Get(name, emptyGetOptions)
+	bp2, err := cli.Blueprints(s.namespace).Get(ctx, name, emptyGetOptions)
 	c.Assert(err, IsNil)
 	c.Assert(bp1, DeepEquals, bp2)
 
 	bp2.Actions = map[string]*crv1alpha1.BlueprintAction{}
-	bp3, err := cli.Blueprints(s.namespace).Update(bp2)
+	bp3, err := cli.Blueprints(s.namespace).Update(context.TODO(), bp2, metav1.UpdateOptions{})
```
Comment (pavannd1) [inline pkg/resource/resource_test.go:129]:
```suggestion
	bp3, err := cli.Blueprints(s.namespace).Update(ctx, bp2, metav1.UpdateOptions{})
```

File: pkg/resource/resource_test.go [hunk 8]
```diff
@@ -117,27 +117,27 @@ func (s *ResourceSuite) TestBlueprintClient(c *C) {
...
 	c.Assert(err, IsNil)
 	c.Assert(bp1.Actions, IsNil)
 	c.Assert(bp3.Actions, NotNil)
 
-	bp4, err := cli.Blueprints(s.namespace).Get(name, emptyGetOptions)
+	bp4, err := cli.Blueprints(s.namespace).Get(context.TODO(), name, emptyGetOptions)
```
Comment (pavannd1) [inline pkg/resource/resource_test.go:134]:
```suggestion
	bp4, err := cli.Blueprints(s.namespace).Get(ctx, name, emptyGetOptions)
```

File: pkg/resource/resource_test.go [hunk 9]
```diff
@@ -117,27 +117,27 @@ func (s *ResourceSuite) TestBlueprintClient(c *C) {
...
 	c.Assert(err, IsNil)
 	c.Assert(bp4, DeepEquals, bp3)
 
-	err = cli.Blueprints(s.namespace).Delete(name, nil)
+	err = cli.Blueprints(s.namespace).Delete(context.TODO(), name, metav1.DeleteOptions{})
```
Comment (pavannd1) [inline pkg/resource/resource_test.go:138]:
```suggestion
	err = cli.Blueprints(s.namespace).Delete(ctx, name, metav1.DeleteOptions{})
```

File: pkg/resource/resource_test.go [hunk 10]
```diff
@@ -117,27 +117,27 @@ func (s *ResourceSuite) TestBlueprintClient(c *C) {
...
 	c.Assert(err, IsNil)
 
-	_, err = cli.Blueprints(s.namespace).Get(name, emptyGetOptions)
+	_, err = cli.Blueprints(s.namespace).Get(context.TODO(), name, emptyGetOptions)
```
Comment (pavannd1) [inline pkg/resource/resource_test.go:141]:
```suggestion
	_, err = cli.Blueprints(s.namespace).Get(ctx, name, emptyGetOptions)
```

File: pkg/controller/controller.go [hunk 11]
```diff
@@ -122,13 +122,13 @@ func (c *Controller) StartWatch(ctx context.Context, namespace string) error {
...
+	if _, err := cli.CrV1alpha1().ActionSets(ns).List(context.TODO(), v1.ListOptions{}); err != nil {
 		return errors.Wrap(err, "Could not list ActionSets")
 	}
-	if _, err := cli.CrV1alpha1().Blueprints(ns).List(v1.ListOptions{}); err != nil {
+	if _, err := cli.CrV1alpha1().Blueprints(ns).List(context.TODO(), v1.ListOptions{}); err != nil {
 		return errors.Wrap(err, "Could not list Blueprints")
 	}
-	if _, err := cli.CrV1alpha1().Profiles(ns).List(v1.ListOptions{}); err != nil {
+	if _, err := cli.CrV1alpha1().Profiles(ns).List(context.TODO(), v1.ListOptions{}); err != nil {
 		return errors.Wrap(err, "Could not list Profiles")
```
[the remark this answers is not in the document]
Comment (viveksinghggits) [reply pkg/controller/controller.go:132]: try to do this for every other occurance.

[the remark this answers is not in the document]
Comment (A-kanksh-a) [reply pkg/controller/controller.go:132]: @viveksinghggits I agree we can do this. we haven't handled this.

**Pull request merged**